### ISSUE TITLE
GIGSEA

### ISSUE THREAD

**Issue by zhushijia** [Bioconductor/Contributions#720]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/zhushijia/GIGSEA

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For help with submitting your package, please subscribe and post questions
to the [bioc-devel][4] mailing list.

[1]: https://bioconductor.org/developers/package-guidelines/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/

**Comment by LiNk-NY** [comment]
Hi Shijia, @zhushijia 
Thank you for your submission. I will be reviewing your package later today. 
Best regards,
Marcel

**Comment by lgeistlinger** [comment]
Hi @zhushijia 

I guess @LiNk-NY will give you some more formal feedback on code and
implementation practice, I rather took a look into your `DESCRIPTION` file and
your vignette.

It seems as if you have implemented useful functionality for gene set analysis
of GWAS data, but from your documentation / vignette it's currently hard to
understand what you did exactly.

* DESCRIPTION

** Description

Resembles an Abstract of a scientific article.
For a software package, you typically want that to be more compact and generally
comprehensible.
Good practice is a sentence on background, 1-2 sentences on what you have
implemented, and a sentence how this integrates with existing functionality.
A good example is the Description field here: http://bioconductor.org/MultiAssayExperiment.

* vignette

** vignette building instructions are needed in yaml header of Rmd file

** Abstract:
*** Many specific undefined terms, e.g.
- GWAS-and-eQTL-imputed differential gene expression
- GWAS summary data
- ...

*** Your grammar also indicates that you are having a manuscript somewhere but
    a reference is missing:
- We presented the Genotype Imputed (where?)
- We have shown the appropriate type I error rates (where?)

*** Appropriately addresses such challenges for SNP enrichment as gene size,
gene boundary, and multiple-marker regulation? (Why is this challenging and how does it address that?)

** Quick start:

*** Linking

https://cloud.hakyimlab.org/static/img/PrediXcan-GWAS-Summary-PrediXcan.png

here would give some context and help users to understand what you are talking
about.

*** It's not a good idea to comment the command in your quick start. That's
basically an `eval=FALSE` chunk and undermines the reproducibility of your
vignette (see http://bioconductor.org/developers/package-guidelines/#vignettes).
In your case, I would also clearly separate functionality based on MetaXcan
(thus relying on external software) from your subsequent enrichment analysis,
and not combine this into one step.

*** I like the distinction between discrete and continuous gene sets and their representation in sparse matrices

*** User self-defined gene sets --> user-defined gene sets

*** Section 5.1: it's basically unclear what you do here.
Again, do you have any manuscript here that you can refer to or can you refer to 
the literature for similar approaches? If not, explain in more detail what you do. 

** link or include cited references in vignette

**Comment by zhushijia** [comment]
Hi @lgeistlinger 

  Thank you very much for your comments. All make sense. I will try to address. :-)

Best wishes
Shijia

**Comment by LiNk-NY** [comment]
Hi Shijia Zhu, @zhushijia 
Thank you for your submission.
Please see the review for GIGSEA below.

If you have any questions please post them here. 
Thanks!
-Marcel 

---

# GIGSEA #720

* Please consider using formatR or a programmatic editor to format your files
to not exceed 80 column-width. This includes your `README.md`, `DESCRIPTION`,
etc. This would make your files easier to maintain and review.
* Remove any `.DS_Store` files from the repository and add the name to the
`.gitignore` file.
* Remove the `*.Rproj` file

## README.md

* You likely don't need to provide a `wget` command for installing your
package. Let users make use of `devtools` or `remotes` for downloading and
installing `GIGSEA` in the meantime.

## DESCRIPTION

* Consider selectively importing packages. You should only include packages
you are extending in the `Depends` field.
* Add these packages to the `Imports` field and selectively import functions with
`@importFrom`.

## NAMESPACE

* Make sure you are consistent with class names (e.g., `dataframe` instead of
`dataFrame`). Consider using the `S4Vectors::DataFrame` as well.
* Try to shorten some names for example, `permutationMultipleLm` could be
`permuteMultiLM` or something similar.
* You may want to name your main function the same as your package name
`GIGSEA`.

## R

* Check the `BiocCheck` results and follow the recommendations for avoiding
`sapply` etc.
* Is it necessary to include all of these datasets in the package? Do other
packages already provide this data?
* Ensure that your argument descriptions are making sense, for example in
`termCol`, is it an index the split vector?
* Avoid nesting for loops where possible.
* Use the allocate and fill method instead of copy and append. For example,
use `empiricalPval <- vector("numeric", nrow(shuffledPval))`
* Avoid nesting lists as well. There are more efficient ways of representing
the data
* Use helper functions to avoid repetitive code
* Allow options to suppress print outs to the screen with arguments like
`verbose = FALSE`
* Use `message()` instead of `cat()` so these messages can be suppressed if
desired by the user.
* You shouldn't have to use `setwd` inside a function. If you have intermediate
files, you can create them in a `tempdir()`. 

Minor:
* Use the arrow `<-` for assignment

## vignettes

* Add the `vignette:` field in your RMarkdown document like this:

```
vignette: >
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteIndexEntry{GIGSEA: Genotype Imputed Gene Set Enrichment Analysis}
  %\VignetteEncoding{UTF-8}
```

## inst/NEWS

* Update your news file

---

**Comment by LiNk-NY** [comment]
Thanks Ludwig @lgeistlinger  for the scientific content review! :+1:

**Comment by zhushijia** [comment]
Hi LiNk-NY @LiNk-NY 

  Thank you very much for the detailed comments. I will try my best to address them. 

Best wishes
Shijia

**Comment by LiNk-NY** [comment]
Please bump your package's version for another build in the BBS/SPB. 
Thanks,
Marcel

**Comment by LiNk-NY** [comment]
Hi Shijia,

Please respond to the review and push your changes. Otherwise I will have to close the issue within 2 weeks. 

Thank you for your attention. 

Regards,
Marcel

**Comment by zhushijia** [comment]
Hi  Marcel,

   I apologize for the late response. I was wrapping up the corresponding
paper recently, and eventually submit it.
I will address the comments and respond quickly.

  Thank you so much for your patient help.

Best Wishes
Shijia

**Comment by LiNk-NY** [comment]
Hi ShiJia, @zhushijia

Please remove all `.DS_Store` files as they are an artifact of your OS.

```
ERROR: System Files found that should not be git tracked:
  GIGSEA.Rproj
  .DS_Store
```

It also seems that `*.Rproj` file instances are returning errors.
I will investigate further.

Thanks,
Marcel

**Comment by zhushijia** [comment]
Hi LiNk-NY, @LiNk-NY, 

  Thank you so much for your help! 

   I have some NOTES that need to address. One of them is how to format the vignettes Rmd file to not exceed 80-column width. Seems simply "Enter" does not work. So, may I ask how to format?

Thank you very much!
Best
Shijia 

    * NOTE: Consider shorter lines; 48 lines (2%) are > 80 characters
      long.
    First 6 lines:
      R/dataFrame2geneSet.R:37         genes = vapply( strsplit( x , ' ' ) , ...
      R/dataFrame2geneSet.R:41         values = vapply( strsplit( x , ' ' ) ,...
      vignettes/GIGSEA_tutorial.Rmd:5   | Department of Genetics and Genomic ...
      vignettes/GIGSEA_tutorial.Rmd:23 We presented the Genotype Imputed Gene...
      vignettes/GIGSEA_tutorial.Rmd:27 In GIGSEA, the gene sets are saved as ...
      vignettes/GIGSEA_tutorial.Rmd:34 GIGSEA first uses MetaXcan (also calle..

**Comment by LiNk-NY** [comment]
Hi Shijia, @zhushijia 

If you can't shorten those lines in the `yaml` front matter, don't worry about it. 
You should be able to shorten other lines in the vignette though.
This will improve readability and ease of maintainership. 

Regards,
Marcel

**Comment by zhushijia** [comment]
Hi LiNk-NY, @LiNk-NY,
  Got it! Thank you very much!
Best
Shijia

**Comment by LiNk-NY** [comment]
Hi ShiJia, @zhushijia 

Please remove the `.Rproj.user` folder from the repository. 
It seems like you accidentally deleted your `.gitignore` file and
commited files that were supposed to be ignored. Please re-add this file.

Your package is close to being accepted. Thank you.

Regards,
Marcel

**Comment by zhushijia** [comment]
Hi LiNk-NY, @LiNk-NY
  
As suggested, I already removed the .Rproj.user folder, and added the .gitignore file.
Thank you very much for your patient help!

Best
Shijia

**Comment by LiNk-NY** [comment]
Hi ShiJia, @zhushijia 
Thanks for making those changes. 
Your package has been accepted. 

Best regards,
Marcel

**Comment by mtmorgan** [comment]
The master branch of your GitHub repository has been added to Bioconductor's git repository.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/zhushijia.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `biocLite(\"GIGSEA\")`. The package 'landing page' will be created at

https://bioconductor.org/packages/GIGSEA

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/

**Comment by zhushijia** [comment]
Hi Martin,

  Yes, it has ssh public keys. Thank you very much for your help!

Best
Shijia

**Comment by zhushijia** [comment]
Hi Marcel,

  I would like to submit the corresponding experimental data for the GIGSEA
package: https://github.com/zhushijia/GIGSEAdata
But the submission of GIGSEA has been accepted, and the issue has been
closed. So, may I ask whether I should open a new issue for the GIGSEAdata?

Thank you very much!
Best
Shijia

**Comment by mtmorgan** [comment]
Yes I suggest simply opening another issue.

**Comment by LiNk-NY** [comment]
Hi Shijia, @zhushijia 
Sorry I missed your message. 
It would've been better to submit both packages in one issue but now that the issue is closed.
Please create another issue and submit it there. 
Thanks!
Regards,
Marcel

**Comment by zhushijia** [comment]
Got it! Thank you very much, Martin.

**Comment by zhushijia** [comment]
Hi Marcel, no problem. Got it! Thank you very much!
Best
Shijia
